### PR TITLE
feat(rule): add use-injectable-provided-in

### PR DESCRIPTION
**Pull request by mohammedzamakhan**
https://github.com/mgechev/codelyzer/issues/558

### CODE REVIEW

File: test/useProvidedInRule.spec.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { assertAnnotated, assertSuccess } from './testHelper';
+
+const {
+  metadata: { ruleName },
+  FAILURE_STRING
+} = Rule;
+
+describe(ruleName, () => {
+  describe('failures', () => {
+    it('should fail if providedIn property is not passed', () => {
```
Comment (rafaelss95) [inline test/useProvidedInRule.spec.ts]:
```suggestion
    it('should fail if providedIn property is not set', () => {
```

File: test/useProvidedInRule.spec.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      const source = `
+        @Injectable({
+          providedIn: 'root'
+        })
+        class Test {}
+      `;
+      assertSuccess(ruleName, source);
+    });
+
+    it('should succeed if providedIn property is set to a property', () => {
```
Comment (rafaelss95) [inline test/useProvidedInRule.spec.ts]:
```suggestion
    it('should succeed if providedIn property is set to a module', () => {
```

**Comment by rafaelss95** [comment]
@mohammedzamakhan I've left some minor comments, please have a look.

**Comment by rafaelss95** [comment]
@mohammedzamakhan Can you just change the commit message to `feat(rule): add use-injectable-provided-in`?

**Comment by mohammedzamakhan** [comment]
@rafaelss95 done 👍

**Comment by mgechev** [comment]
I'll restart travis and merge when the build passes.

**Comment by rafaelss95** [comment]
I investigated the reason behind the build failure and it seems that the problem is with `node-sass` dependency: our _travis_ config (as you may already know) currently runs with Node 8.11.1 and Node 12 (`latest stable`), however, [node-sass is not compatible with version 12 node](https://github.com/sass/node-sass/issues/2632).

The fix should have come in https://github.com/sass/node-sass/commit/23c86596a9bbcf1e72c6871e3a7cf084d7b68496, [but apparently it hasn't yet been published in npm](https://www.npmjs.com/package/node-sass).

**Pull request merged**